catalog/admin/module/catalog-admin-module-maplayers/src/main/webapp/map-layers/index.js

```diff
@@ -0,0 +1,263 @@
+        <Flexbox justifyContent='center' alignItems='center' width='100%'>
+          <CircularProgress size={60} thickness={7} />
+        </Flexbox>
+      </div>
+      : null}
+    {children}
+  </div>
+)
+
+const Title = muiThemeable()(({ children, muiTheme }) => (
+  <h1 style={{ color: muiTheme.palette.textColor, textAlign: 'center' }}>{children}</h1>
+))
+
+const implementationSupport = (implementors) => {
+  if (!implementors.includes('ol')) {
+    return '3D Only'
+  } else if (!implementors.includes('cesium')) {
+    return '2D Only'
+  }
+  return null
+}
+
+let Error = ({ muiTheme: { palette }, errorText, children }) => (
+  <div>
+    <div style={{
+      border: ('2px solid ' + (errorText !== undefined ? palette.errorColor : palette.disabledColor)),
+      borderRadius: 2
+    }}>
+      {children}
+    </div>
+    <div style={{ color: palette.errorColor, marginTop: 8, fontSize: '0.8em' }}>{errorText}</div>
+  </div>
+)
+
+Error = muiThemeable()(Error)
+
+const ProviderEditor = ({ provider, error = Map(), onUpdate }) => (
+  <div style={{ padding: 16 }}>
+    <div style={{ padding: '0 16px' }}>
+      <TextField
+        onChange={(e, value) => onUpdate(value.replace(/ /g, ''), 'url')}
+        fullWidth
+        errorText={error.get('url')}
+        value={provider.get('url') || ''}
+        id='provider-url'
+        floatingLabelText='Provider URL' />
+    </div>
+    <Flexbox flex='1' style={{ padding: '0 16px' }}>
+      <Flexbox flex='5' style={{ marginRight: 20 }}>
+        <div style={{ width: '100%' }}>
+          <SelectField
+            floatingLabelText='Provider Type'
+            fullWidth
+            value={provider.get('type') || ''}
+            errorText={error.get('type')}
+            onChange={(e, i, value) => onUpdate(value, 'type')}>
+            {Object.keys(options).map((type, i) =>
+              <MenuItem
+                key={i}
+                value={type}
+                primaryText={
+                  <Flexbox flex='1' justifyContent='space-between'>
+                    <div>{options[type].label}</div>
+                    <div>
+                      {implementationSupport(Object.keys(options[type].help))}
+                    </div>
+                  </Flexbox>} />)}
+          </SelectField>
+        </div>
+      </Flexbox>
+      <Flexbox flex='1'>
+        <TextField
+          type='number'
+          step={0.01}
+          value={typeof provider.get('alpha') === 'number' ? provider.get('alpha') : ''}
+          errorText={error.get('alpha')}
+          floatingLabelText='Alpha'
+          fullWidth
+          onChange={(e, value) => {
+            if (value === '') {
+              onUpdate('', 'alpha')
+            } else {
+              const n = Number(value)
+              if (!(n < 0 || n > 1)) {
+                onUpdate(n, 'alpha')
+              }
+            }
+          }} />
+      </Flexbox>
+    </Flexbox>
+    <List>
+      <ListItem
```

Comment:
The  ListItem attribute `primaryTogglesNestedList` will allow the entire list item to be clickable instead of just the up/down arrow on the right